scripts/pam-test.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -e
+
+# load common functions
+# shellcheck disable=SC1091
+. ./bash_lib.sh
+
+bb_print_env
+
+if ! which mysql ; then
+  bb_log_err "This step assumes that MariaDB has already been installed"
+  exit 1
+fi
+
+control_mariadb_server restart
+
+set +e
+
+res=0
+
+#----------------
+# Basic pam_unix
+#----------------
+
+set -e
+
+sudo tee /etc/pam.d/mariadb <<EOF
+auth required pam_unix.so audit
+account required pam_unix.so audit
+EOF
+
+# PAM v2
+
+sudo mysql -e "INSTALL SONAME 'auth_pam'; CREATE USER 'buildbot'@'localhost' IDENTIFIED VIA pam USING 'mariadb'"
+if ! mysql -ubuildbot -ptest -e "SHOW GRANTS" ; then
+  res=1
+  bb_log_err "Authentication with PAM v2 (pam_unix) failed"
+fi
+sudo mysql -e "UNINSTALL SONAME 'auth_pam'"
+if mysql -ubuildbot -ptest -e "SHOW GRANTS" > /dev/null 2>&1 ; then
+  res=1
+  bb_log_err "User authenticated via PAM v2 (pam_unix) could still connect after uninstalling plugin"
+fi
+
+if [ "$res" == "0" ] ; then
+  bb_log_info "PAM v2 Authentication test successful"
+fi
+
+# PAM v1
+
+sudo mysql -e "INSTALL SONAME 'auth_pam_v1'"
+
+set +e
+sudo groupadd shadow
+sudo usermod -a -G shadow mysql
+sudo chown root:shadow /etc/shadow
+sudo chmod g+r /etc/shadow
+set -e
+
+control_mariadb_server restart
+
+if ! mysql -ubuildbot -ptest -e "SHOW GRANTS" ; then
+  res=1
+  bb_log_err "Authentication with PAM v1 (pam_unix) failed"
+fi
+sudo mysql -e "UNINSTALL SONAME 'auth_pam_v1'"
+if mysql -ubuildbot -ptest -e "SHOW GRANTS" > /dev/null 2>&1 ; then
+  res=1
+  bb_log_err "User authenticated via PAM v1 (pam_unix) could still connect after uninstalling plugin"
+fi
+
+if [ "$res" == "0" ] ; then
+  bb_log_info "PAM v1 Authentication test successful"
+fi
+
+#----------------
+# MTR
+#----------------
+
+cd /usr/share/mysql-test || cd /usr/share/mariadb-test || cd /usr/share/mysql/mysql-test || cd /usr/share/mariadb/mariadb-test
```

Comment:
hmm this is confusing an does not look clean, what about something like:
```bash
# PATH can differ between MariaDB versions and distros
for db in mysql mariadb; do
  for dir in \
    /usr/share/$db-test \
    /usr/share/$db/$db-test; do
    [[ -d "$dir" ]] && cd "$dir"
  done
done
```

Reply:
Sure, I'll try it out. Much cleaner, thanks!